addons/website/views/snippets/snippets.xml

```diff
@@ -1139,22 +1139,33 @@
             <we-button data-customize-website-variable="'postcard'"><span><img src="/website/static/src/img/snippets_options/page_layout_postcard.svg"/></span><span class="pl-2">Postcard</span></we-button>
         </we-select>
-        <we-row string="⌙ Background" data-no-preview="true">
+
+        <we-row string="⌙ Background" data-no-preview="true" class="o_we_full_row">
             <we-colorpicker data-dependencies="!layout_full_opt"
-                            data-customize-website-color=""
-                            data-color="body"/>
-            <we-button-group data-imagepicker="body_bg_image_opt">
-                <we-button title="Image"
-                           data-customize-body-bg-type="'image'"><i class="fa fa-fw fa-camera"/></we-button>
-                <we-button title="Pattern"
-                           data-customize-body-bg-type="'pattern'"><i class="fa fa-fw fa-th"/></we-button>
-                <we-button title="None"
-                           data-customize-body-bg-type="NONE"><i class="fa fa-fw fa-ban"/></we-button>
-            </we-button-group>
-            <!-- Hidden imagepicker enabled by above button-group -->
-            <we-imagepicker data-name="body_bg_image_opt"
-                            data-customize-body-bg=""/>
+                    data-customize-website-color="" data-color="body"/>
+            <we-button title="Image" class="ml-auto" data-name="bg_image_toggle_opt"
+                    data-toggle-body-bg-image="true" data-no-preview="true">
+                <i class="fa fa-fw fa-camera"/>
+            </we-button>
+        </we-row>
+        <we-row string=" ⌙ Image"><!-- &emsp; -->
+            <we-imagepicker title="Edit image" data-dependencies="bg_image_toggle_opt" data-name="body_bg_image_opt" data-customize-body-bg=""/>
+        </we-row>
+        <we-row string=" ⌙ Position"><!-- &emsp; -->
+            <we-select data-no-preview="true" data-variable="body-image-type" data-dependencies="bg_image_toggle_opt">
+                <we-button data-customize-website-variable="'image'">Cover</we-button>
+                <we-button data-customize-website-variable="'pattern'" data-name="background_repeat_opt">Repeat pattern</we-button>
+            </we-select>
+            <we-button title="Background Position" data-dependencies="bg_image_toggle_opt"
+                    data-background-position-overlay="true" data-no-preview="true">
+                <i class="fa fa-fw fa-crosshairs"/>
+            </we-button>
         </we-row>
+        <we-multi data-dependencies="background_repeat_opt" data-variable="body-image-repeat-size">
+            <we-input string="  ⌙ Width" placeholder="auto" data-unit="px" data-customize-website-variable="auto"/><!-- &emsp; -->
+            <we-input string="  ⌙ Height" placeholder="auto" data-unit="px" data-customize-website-variable="auto"/><!-- &emsp; -->
+        </we-multi>
```

Comment:
Because the touched parameters themselves are different, I defined both families with `t-att-*`s. A parameter now indicates which family of parameters to use.